src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,194 @@
+    private int tableSize = 0;
+    private int threshold;
+
+    static class Node<K, V> {
+        private final int hash;
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+
+        public int getHash() {
+            return hash;
+        }
+
+        public K getKey() {
+            return key;
+        }
+
+        public V getValue() {
+            return value;
+        }
+
+        public Node<K, V> getNext() {
+            return next;
+        }
+
+        public void setValue(V value) {
+            this.value = value;
+        }
+
+        public void setNext(Node<K, V> next) {
+            this.next = next;
+        }
+
+        public final int hashCode() {
+            return Objects.hashCode(key) ^ Objects.hashCode(value);
+        }
+
+        public final boolean equals(Object current) {
+            if (current == this) {
+                return true;
+            }
+
+            if (current == null) {
+                return false;
+            }
+
+            return current instanceof Node<?, ?> origin
+                    && Objects.equals(key, origin.getKey())
+                    && Objects.equals(value, origin.getValue());
+        }
+    }
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
+
+    public final int hash(Object key) {
+        int hash;
+        return (key == null) ? 0 : (hash = key.hashCode()) ^ (hash >>> 16);
+    }
+
+    final Node<K, V>[] resize() {
+        Node<K, V>[] oldTable = table;
+        int oldCapacity = (oldTable == null) ? 0 : oldTable.length;
+        int oldThreshold = threshold;
+        int newCapacity;
+        int newThreshold = 0;
+
+        if (oldCapacity > 0) {
+            if (oldCapacity >= MAXIMUM_CAPACITY) {
+                threshold = Integer.MAX_VALUE;
+                return oldTable;
+            } else if ((newCapacity = oldCapacity << 1) < MAXIMUM_CAPACITY
+                    && oldCapacity >= DEFAULT_INITIAL_CAPACITY) {
+                newThreshold = oldThreshold << 1;
+            }
+        } else if (oldThreshold > 0) {
+            newCapacity = oldThreshold;
+        } else {
+            newCapacity = DEFAULT_INITIAL_CAPACITY;
+            newThreshold = (int) (DEFAULT_LOAD_FACTOR * DEFAULT_INITIAL_CAPACITY);
+        }
+
+        if (newThreshold == 0) {
+            float resizeThreshold = (float) newCapacity * 0.75f;
+            newThreshold = (newCapacity < MAXIMUM_CAPACITY
+                    && resizeThreshold < (float) MAXIMUM_CAPACITY
+                    ? (int) resizeThreshold : Integer.MAX_VALUE);
+        }
+        threshold = newThreshold;
+
+        @SuppressWarnings({"rawtypes", "unchecked"})
+        Node<K, V>[] newTable = (Node<K, V>[]) new Node[newCapacity];
```

Comment:
Be attentive with the diamond operator in Java. Use `new Node<K, V>[newCapacity]` instead of `new Node[newCapacity]` to avoid unchecked assignment warnings.